src/systemctl/systemctl-edit.c

```diff
@@ -374,7 +389,6 @@ int verb_edit(int argc, char *argv[], void *userdata) {
                         /* No need to check if the dir is empty, rmdir does nothing if it is not the case. */
                         (void) rmdir(dir);
                 }
-        }
 
-        return 0;
+        return r > 0 ? 0 : r;
```

Comment:
Hm? Why the change is necessary? When we reach here, then there is no error we should propagate. Right?